packages/next/client/image.tsx

```diff
@@ -374,6 +378,18 @@ export default function Image({
         `Image with src "${src}" has both "priority" and "loading='lazy'" properties. Only one should be used.`
       )
     }
+    if (placeholder === 'blur') {
+      if ((widthInt || 0) * (heightInt || 0) < 1600) {
+        console.warn(
+          `Image with src "${src}" is smaller than 40x40. Consider removing the "placeholder='blur'" property to improve performance.`
+        )
+      }
+      if (!blurDataURL) {
+        throw new Error(
+          `Image with src "${src}" has "placeholder='blur'" property but is missing the "blurDataURL" property.`
```

Comment:
Should we mention that this property is needed for non-statically-imported images?

Reply:
@atcastle The user won't get this error if they have a statically imported image. Well now that I think about, they might if the image is not png or jpeg. So should we suggest that they use a supported file type?

Reply:
`width`/`height` only should still work, but if `placeholder="blur"` and using an unsupported image should throw 👍